lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -333,27 +338,54 @@ public Builder withClaim(String name, Map<String, ?> map) throws IllegalArgument
         /**
          * Add a custom List Claim with the given items.
          * <p>
-         * Accepted nested types are {@linkplain Map} and {@linkplain List} with basic types
-         * {@linkplain Boolean}, {@linkplain Integer}, {@linkplain Long}, {@linkplain Double},
-         * {@linkplain String} and {@linkplain Date}. {@linkplain Map}s cannot contain null keys or values.
-         * {@linkplain List}s can contain null elements.
+         * Accepted nested types are {@linkplain Map} and {@linkplain List} with basic
+         * types {@linkplain Boolean}, {@linkplain Integer}, {@linkplain Long},
+         * {@linkplain Double}, {@linkplain String} and {@linkplain Date}.
+         * {@linkplain Map}s cannot contain null keys or values. {@linkplain List}s can
+         * contain null elements.
          *
          * @param name the Claim's name.
          * @param list the Claim's list of values.
          * @return this same Builder instance.
-         * @throws IllegalArgumentException if the name is null, or if the list contents does not validate.
+         * @throws IllegalArgumentException if the name is null, or if the list contents
+         *                                  does not validate.
          */
 
         public Builder withClaim(String name, List<?> list) throws IllegalArgumentException {
             assertNonNull(name);
             // validate list contents
             if (list != null && !validateClaim(list)) {
-                throw new IllegalArgumentException("Expected list containing Map, List, Boolean, Integer, Long, Double, String and Date");
+                throw new IllegalArgumentException(
+                        "Expected list containing Map, List, Boolean, Integer, Long, Double, String and Date");
             }
             addClaim(name, list);
             return this;
         }
 
+        /**
+         * Add specific Claims to set as the Payload. If provided map is null then
+         * nothing is changed If provided map contains a claim with null value then that
+         * claim will be removed from the payload
+         *
+         * @param payloadClaims the values to use as Claims in the token's payload.
+         * @return this same Builder instance.
+         */
+        public Builder withPayload(Map<String, Object> payloadClaims) {
+            if (payloadClaims == null) {
+                return this;
+            }
+
+            for (Map.Entry<String, Object> entry : payloadClaims.entrySet()) {
+                if (entry.getValue() == null) {
+                    this.payloadClaims.remove(entry.getKey());
+                } else {
+                    this.payloadClaims.put(entry.getKey(), entry.getValue());
+                }
+            }
+
+            return this;
+        }
```

Comment:
We should perform the same claim validations here as we do for others (key not null, values are supported types (should be able to use the `isSupportedType() method`).

Reply:
Yeah, I wasn't sure about that. But I can add `isSupportedType()`, I will check it later on